src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -432,187 +867,194 @@ class npc_memory : public CreatureScript
         npc_memoryAI(Creature* creature) : ScriptedAI(creature)
         {
             Initialize();
+            me->SetReactState(REACT_PASSIVE);
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+            _events.ScheduleEvent(EVENT_OLD_WOUNDS, urand(14 * IN_MILLISECONDS, 16 * IN_MILLISECONDS));
+            _events.ScheduleEvent(EVENT_SHADOWS_PAST, 8 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_WAKING_NIGHTMARE, urand(10 * IN_MILLISECONDS, 13 * IN_MILLISECONDS));
         }
 
         void Initialize()
         {
-            uiOldWoundsTimer = 12000;
-            uiShadowPastTimer = 5000;
-            uiWakingNightmare = 7000;
-        }
+            DoCastAOE(SPELL_SHADOWFORM);
+            DoCastAOE(SPELL_SPAWN_VISUAL);
 
-        uint32 uiOldWoundsTimer;
-        uint32 uiShadowPastTimer;
-        uint32 uiWakingNightmare;
+        }
 
         void Reset() override
         {
+            _events.Reset();
```

Comment:
Will there ever be a reasonable `Reset` invocation? If no, why does this still exist? If yes, when?